### PR TITLE
Invert isinstance logic

### PR DESCRIPTION
**Pull request by labrys**
`isinstance` should be checking to see if the instance is a string type not a list type.  Checking for list type is a bug.  For example:

```python
# Current behaviour

>>> arg = 'Some Text'
>>> [arg] if not isinstance(arg, list) else arg
['Some Text']  # Expected result so we don't iterate through the characters of the string

>>> arg = (a, b, c)  # but with this...
>>> [arg] if not isinstance(arg, list) else arg
[(a, b, c)]  # Tuple inside a list is a bug

# New behaviour

>>> arg = 'Some Text'
>>> [arg] if isinstance(arg, string_types) else arg
['Some Text']  # Expected result

>>> arg = (a, b, c)  # now with this...
>>> [arg] if isinstance(arg, string_types) else mapped_results
(a, b, c)  # Still an iterable, bug is fixed
```

### CODE REVIEW

**Comment by fernandog** [comment]
@labrys what about all this others?

image

**Comment by labrys** [comment]
@fernandog They still need to be done.  These were meant to be a part of p0psicles previous PR as they were the ones directly related to it but it got merged before I got to add these edits.

File: medusa/indexers/tvdbv2/tvdbv2_api.py [hunk 1]
```diff
@@ -104,7 +105,7 @@ def _object_to_dict(self, tvdb_response, key_mapping=None, list_separator='|'):
 
         tvdb_response = getattr(tvdb_response, 'data', tvdb_response)
 
-        if not isinstance(tvdb_response, list):
+        if isinstance(tvdb_response, string_types):
```
Comment (p0psicles) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: tvdb_response is not a string, it's an object or list of objects.

File: medusa/indexers/tvdbv2/tvdbv2_api.py [hunk 2]
```diff
@@ -180,7 +181,7 @@ def search(self, series):
             return
 
         mapped_results = self._object_to_dict(results, self.series_map, '|')
-        mapped_results = [mapped_results] if not isinstance(mapped_results, list) else mapped_results
+        mapped_results = [mapped_results] if isinstance(mapped_results, string_types) else mapped_results
```
Comment (p0psicles) [inline medusa/indexers/tvdbv2/tvdbv2_api.py]: not a string, but OrderedDict or list of OrderedDict

**Comment by p0psicles** [comment]
there are more changes done here, that will simply break when adding a show.
Please look carefully at each evaluated isinstance, as most of them are not strings.

**Comment by p0psicles** [comment]
@labrys are you planning on finishing this?

**Comment by labrys** [comment]
Yes I am but this is a low priority for me.  If you or anyone else wants to pick up on it I'm not opposed.

**Comment by labrys** [comment]
Closing as this is a low priority for me right now and don't have time to commit to it.

**Pull request closed without merging**